src/app/components/user-details/user-details.component.ts

```diff
@@ -0,0 +1,126 @@
+
+    tempDetails = {
+        name: "Username",
+        position: "Position"
+    }
+
+    //until getCurrentlyLoggedUser call is implemented
+    userId: number = 7;
+
+    get controls() {
+        return this.userEditForm.controls
+    }
+
+    constructor(private userService: UserService, private _snackBar: MatSnackBar) { }
+
+    ngOnInit(): void {
+        this.initUserFormControls();
+        this.retrieveUserData();
+    }
+
+    initUserFormControls(): void {
+        this.userEditForm = new FormGroup({
+            name: new FormControl({ value: '', disabled: true }, [Validators.required, Validators.pattern('^[a-zA-Z]+[a-zA-Z ]*')]),
+            position: new FormControl({ value: '', disabled: true }, [Validators.required]),
+            phoneNo: new FormControl({ value: '', disabled: true }, [Validators.pattern('^[0-9]{0,10}')]),
+            email: new FormControl({ value: '', disabled: true }, null)
+        });
+    }
+
+    retrieveUserData(): void {
+        this.loading = true;
+        this.userService.getUserById(this.userId).subscribe(
+            (result) => {
+
+                this.tempDetails.name = result.name;
+                this.tempDetails.position = result.position;
+
+                this.controls.name.setValue(this.ifValueEmpty(result.name));
+                this.controls.position.setValue(this.ifValueEmpty(result.position));
+                this.controls.phoneNo.setValue(this.ifValueEmpty(result.phoneNo));
+                this.controls.email.setValue(this.ifValueEmpty(result.email));
+
+                this.loading = false;
+            },
+            (error) => {
+                {
+                    this._snackBar.open(
+                        `Error status ${error.status}: ${error.message}`,
+                        '',
+                        {
+                            duration: 5000,
+                        }
+                    );
+                }
+            }
+        );
+    }
+
+    ifValueEmpty(initialValue): string {
+        return initialValue == '' ? 'N/A' : initialValue;
+    }
+
+    toggleEditMode(): void {
+        this.editMode = !this.editMode;
+
+        if (this.editMode) {
+            this.controls.name.enable();
+            this.controls.position.enable();
+            this.controls.phoneNo.enable();
+        }
+        else {
+            this.userEditForm.disable();
+        }
+    }
+
+    onSubmit(): void {
+        let user: UserUpdate = {
+            id: this.userId,
+            name: this.controls.name.value,
+            phoneNo: this.controls.phoneNo.value,
+            position: this.controls.position.value,
+        }
+        this.loading = true;
+        this.userService.putUser(user).subscribe(
+            (result) => {
+                this.loading = false;
+
+                this._snackBar.open('User successfully updated.', '', {
+                    duration: 3000,
+                });
+
+                this.retrieveUserData();
```

Comment:
You should not GET again the details of the user. The PUT action should already send the updated user object when the call is successful.  You should make use of that.